src/components/Banner/Banner.tsx

```diff
@@ -0,0 +1,239 @@
+   * Refer to how color values are defined and typed in the DS Icon component. */
+  backgroundColor?: BannerBgColors;
+  /** Additional `className` to add. */
+  className?: string;
+  /** Used to populate the body content of the component. */
+  content: string | JSX.Element;
+  /** Used to populate the heading element within the component.  A string
+   * value can be passed to set the text for a default DS Heading component,
+   * or an actual DS Heading component can be passed in. */
+  heading?: string | JSX.Element;
+  /** Used to set the color for the left border and icon.  Refer to how color
+   * values are defined and typed in the DS Icon component. */
+  highlightColor?: BannerHighlightColors;
+  /** Optional custom `Icon` that will override the default `Icon`. */
+  icon?: JSX.Element;
+  /** ID that other components can cross reference for accessibility purposes. */
+  id?: string;
+  /** Optional prop to control whether a `Banner` can be dismissed
+   * (closed) by a user. */
+  isDismissible?: boolean;
+  /** Used to control the component's semantic coloring and iconography. */
+  type?: BannerTypes;
+}
+
+const iconProps: Record<BannerTypes, IconProps> = {
+  neutral: {
+    name: "errorOutline",
+    title: "Banner neutral icon",
+    iconRotation: "rotate180",
+  },
+  informative: {
+    name: "errorOutline",
+    title: "Banner informative icon",
+    iconRotation: "rotate180",
+  },
+  positive: {
+    name: "actionCheckCircle",
+    title: "Banner positive icon",
+  },
+  negative: {
+    name: "errorOutline",
+    title: "Banner negative icon",
+  },
+  warning: {
+    name: "alertNotificationImportant",
+    title: "Banner warning icon",
+  },
+  recommendation: {
+    name: "actionLightbulb",
+    title: "Banner recommendation icon",
+  },
+};
+
+/**
+ * The `Banner` component is a non-modal semantic dialog used to communicate a
+ * general status event or to promote a feature, providing contextual feedback
+ * messages for typical user actions. They are displayed contextually within a
+ * page flow and they will often prompt a user to take action.
+ */
+export const Banner: ChakraComponent<
+  React.ForwardRefExoticComponent<
+    BannerProps & React.RefAttributes<HTMLDivElement>
+  >,
+  BannerProps
+> = chakra(
+  forwardRef<HTMLDivElement, BannerProps>((props, ref?) => {
+    const {
+      ariaLabel,
+      backgroundColor,
+      className,
+      content,
+      heading,
+      highlightColor,
+      icon,
+      id,
+      isDismissible = false,
+      type = "neutral",
+      ...rest
+    } = props;
+    const [isOpen, setIsOpen] = useState(true);
+    const handleClose = () => setIsOpen(false);
+    const styles = useMultiStyleConfig("Banner", {
+      backgroundColor,
+      highlightColor,
+      // If either `backgroundColor` or `highlightColor` are set, then it
+      // overrides the Banner types.
+      variant: backgroundColor || highlightColor ? undefined : type,
```

Comment:
Not a fan of passing undefined but this works to override the variant if either backgroundColor or highlightColor is passed.

Reply:
I am seeing some issues if `backgroundColor` or `highlightColor` is set without settign the other.  The original idea was for the colors based on the `type` value to be used unless explicitly overridden with these props.  If only one is set, then the default color for the color that was not overridden should still be used.  However, that is not what is happening currently.

Let's talk about this functionality.  Maybe we should make both required if one is set.

Reply:
Hmm so if the variant is "informative" but `highlightColor` is passed as "brand.primary", then it would still have the light blue background and only update the highlightColor?

It's possible but it might me re-doing this so that it uses the `type` prop instead of the `variant` prop, or just redo each variant object.

Reply:
I don't want this to be overly difficult or complex in the code.  Let's make it so the custom colors are required as a pair.  If you set one, you much also set the other.  Conditions and warnings will be necessary.